### PR TITLE
[sitecore-jss] [mediaApi] `updateImageUrl` fix for invalid hash errors on Sitecore

### PR DESCRIPTION
**Pull request by ambrauer**
## Description / Motivation
`updateImageUrl` now *only* switches to JSS media handler prefix (e.g. "/-/jssmedia") if `imageParams` are sent. Otherwise, the original media URL is returned. 

This fixes hash errors ("_ERROR MediaRequestProtection: An invalid/missing hash value was encountered._") from appearing in the Sitecore logs, due to the fact that the prefix is taken into account when calculating the hash value. 

If image size parameters _are_ used, the hash is removed from the image URL and the JSS handler and size whitelisting take over.

## Testing Details
Unit test coverage, tested Next.js and React sample apps.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

### CODE REVIEW

File: packages/sitecore-jss/src/media-api.ts [hunk 1]
```diff
@@ -63,6 +64,10 @@ export const updateImageUrl = (
   params?: { [key: string]: string | number | undefined } | null,
   mediaUrlPrefix: RegExp = mediaUrlPrefixRegex
 ) => {
+  if (!params) {
```
Comment (illiakovalenko) [inline packages/sitecore-jss/src/media-api.ts:67]: I'm wondering, should we check if `params` is empty object? For example using `Object.keys(params).length`? For example I didn't provide any properties, but anyway in this case url will be changed

**Review by illiakovalenko — CHANGES_REQUESTED** [review]
Overall looks good 👍
See few comments below

**Comment by illiakovalenko** [comment]
And, please, pull latest changes from dev - here is conflict

File: packages/sitecore-jss/src/media-api.ts [hunk 2]
```diff
@@ -71,20 +76,15 @@ export const updateImageUrl = (
   }
   const parsed = URL(url, {}, true);
 
-  const query = { ...(params || parsed.query) };
```
[the remark this answers is not in the document]
Comment (ambrauer) [reply packages/sitecore-jss/src/media-api.ts:74]: @illiakovalenko Great catch! Fixed

File: packages/sitecore-jss/src/media-api.ts (hunk 1 above)
Comment (ambrauer) [reply packages/sitecore-jss/src/media-api.ts:67]: @illiakovalenko I went back and forth on this. Had that in there, but then found it was nice as it provides a way to "force" the prefix switch without any parameters (if desired). Useful in unit tests, possibly in the real world as well?

Comment (illiakovalenko) [reply packages/sitecore-jss/src/media-api.ts:67]: @ambrauer But probably, use will define something or will not define anything

Comment (ambrauer) [reply packages/sitecore-jss/src/media-api.ts:67]: @illiakovalenko I'm convinced, we should check object keys b/c of cases where an empty object is passed in by the frameworks usage (and in fact the srcset logic appears to do that). I'll make the change.

Comment (illiakovalenko) [reply packages/sitecore-jss/src/media-api.ts:67]: @ambrauer I can't imagine case when I can specify empty object :)

Comment (illiakovalenko) [reply packages/sitecore-jss/src/media-api.ts:67]: @ambrauer Cool, thank you

Comment (ambrauer) [reply packages/sitecore-jss/src/media-api.ts]: Done

**Review by illiakovalenko — APPROVED** [review]

**Pull request merged**